homeassistant/components/synology_dsm/config_flow.py

```diff
@@ -246,14 +246,31 @@ async def async_step_ssdp(self, discovery_info: DiscoveryInfoType) -> FlowResult
             discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME].split("(", 1)[0].strip()
         )
 
-        mac = discovery_info[ssdp.ATTR_UPNP_SERIAL].upper()
+        discovered_mac = discovery_info[ssdp.ATTR_UPNP_SERIAL].upper()
         # Synology NAS can broadcast on multiple IP addresses, since they can be connected to multiple ethernets.
         # The serial of the NAS is actually its MAC address.
-        if self._mac_already_configured(mac):
-            return self.async_abort(reason="already_configured")
 
-        await self.async_set_unique_id(mac)
-        self._abort_if_unique_id_configured()
+        existing_entry: ConfigEntry | None = None
+        for entry in self.hass.config_entries.async_entries(DOMAIN):
+            if discovered_mac in [
+                mac.replace("-", "") for mac in entry.data.get(CONF_MAC, [])
+            ]:
+                existing_entry = entry
```

Comment:
It would be a bit nicer to repurpose `_mac_already_configured` into `_async_get_existing_entry` or similar.

Also `self._async_current_entries()` is preferred to `self.hass.config_entries.async_entries(DOMAIN)`